upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.6.yaml.template

```diff
@@ -16,10 +16,11 @@ data:
   calico_backend: "bird"
 
   # The CNI network configuration to install on each node.
+  # cniVersion should be 0.1.0 on k8s: https://github.com/projectcalico/calico/issues/742
   cni_network_config: |-
     {
       "name": "k8s-pod-network",
-      "cniVersion": "0.3.0",
+      "cniVersion": "0.1.0",
```

Comment:
Does this work?

I'd expect you to need to also remove the list part here.

For example:

```json
    {
       "name": "k8s-pod-network",
       "cniVersion": "0.1.0",
       "type": "calico",
       "etcd_endpoints": "__ETCD_ENDPOINTS__",
       "log_level": "info",
       "ipam": {
         "type": "calico-ipam"
       },
       "policy": {
         "type": "k8s",
         "k8s_api_root": "https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__",
         "k8s_auth_token": "__SERVICEACCOUNT_TOKEN__"
       },
       "kubernetes": {
         "kubeconfig": "/etc/cni/net.d/__KUBECONFIG_FILENAME__"
       }
     }
```

Reply:
Yep, like this one: https://github.com/kubernetes/kops/pull/3869/files#diff-aefa6a2023a7eddb4e0106cb7c132b96R19

Reply:
I think this got switched because of this: https://github.com/projectcalico/calico/issues/742 could you verify that's still the best version?

Reply:
It depends on the Kubernetes version in use.

Anything less than k8s v1.7 (so v1.5, v1.6) will need CNI version 0.1.0 with the config specified here.

For Kubernets v1.7+, CNI 0.3.0 with the chained CNI config is fine.

Reply:
This is the template for 1.7, let me check the 1.7 template

Reply:
> /networking.projectcalico.org/k8s-1.6.yaml.template

Looks like it's the 1.6 template?

Reply:
@caseydavenport confused.  What am I missing?

Reply:
It could be me that is confused, but AFAICT this is the k8s v1.6 manifest template, but the CNI config has plugin chaining in it which isn't supported until k8s v1.7.

If I'm right, then the config needs to be changed to look like the v1.5. config does.

Reply:
To be clear we need
```yaml
# This ConfigMap is used to configure a self-hosted Calico installation.
kind: ConfigMap
apiVersion: v1
metadata:
  name: calico-config
  namespace: kube-system
data:
  # The calico-etcd PetSet service IP:port
  etcd_endpoints: "{{ $cluster := index .EtcdClusters 0 -}}
                      {{- range $j, $member := $cluster.Members -}}
                          {{- if $j }},{{ end -}}
                          http://etcd-{{ $member.Name }}.internal.{{ ClusterName }}:4001
                      {{- end }}"

# Configure the Calico backend to use.
  calico_backend: "bird"

  # The CNI network configuration to install on each node.
  cni_network_config: |-
    {
        "name": "k8s-pod-network",
        "type": "calico",
        "etcd_endpoints": "__ETCD_ENDPOINTS__",
        "log_level": "info",
        "ipam": {
            "type": "calico-ipam"
        },
        "policy": {
            "type": "k8s",
             "k8s_api_root": "https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__",
             "k8s_auth_token": "__SERVICEACCOUNT_TOKEN__"
        },
        "kubernetes": {
            "kubeconfig": "/etc/cni/net.d/__KUBECONFIG_FILENAME__"
        }
    }
```
ConfigMap in the 1.6 manifest?